### ISSUE TITLE
catalog helm http server should add support to read non-absolute url for .tgzs in index.yaml

### ISSUE THREAD

**Issue by guangbochen** [rancher/rancher#14905]
Rancher Version: master 08/02

Steps to Reproduce:

1. Go to the Global Catalog Menu
2. Add custom HTTP type helm charts server and its index.yaml for charts is using a relative path like `urls: [
"charts/chartmuseum-1.6.0.tgz"
],`
3. the catalog charts server is added but failed to pull the templates, rancher server will showing `unsupported protocol scheme ""` error message, 

```
2018/08/02 09:57:23 [INFO] Updating catalog test
2018/08/02 09:57:23 [INFO] Catalog sync done. 1 templates created, 0 templates updated, 0 templates deleted
2018/08/02 09:57:23 [ERROR] CatalogController test [catalog] failed with : failed to sync templates. Resetting commit. Multiple errors occurred: [Get charts/chartmuseum-1.6.0.tgz: unsupported protocol scheme ""]
2018/08/02 09:57:23 [INFO] Updating catalog test
2018/08/02 09:57:23 [INFO] Catalog sync done. 0 templates created, 0 templates updated, 0 templates deleted
```

Expected Results: need to add support for download template with relative chart URL path by joining the server URL with the chart URL if no absolute path is provided.

**Comment by benzman81** [comment]
Any news on this? We cannot use the nexus plugin https://github.com/sonatype-nexus-community/nexus-repository-helm/ until this is fixed.

**Comment by csturiale** [comment]
It can't be integrated with harbor if it will be not fixed. Thanks

**Comment by artyom-gatalsky** [comment]
workaround: https://github.com/helm/chartmuseum has an ability join server url with chart url

**Comment by daxmc99** [comment]
Should be fixed in https://github.com/rancher/rancher/pull/22369

**Comment by jiaqiluo** [comment]
duplication of https://github.com/rancher/rancher/issues/18535

The bug fix is validated in the above issue and will be available in the coming `Rancher:v2.3` release

**Comment by chris-aeviator** [comment]
In case anybody stumbles upon this and is unsure what to do, if you run your own helm catalog with `chartmuseum`, add the env var `CHART_URL` or start chartmuseum with `--chart_url=http://my-repo.com/`. To validate check if your `http://my-repo.com/index.yaml` contains links like `http://my-repo.com/charts/my-chart.tgz` in the url section
